mongodb/leafygreen-ui
# PD-1308: Allow manual control of active group state for SideNavGroup

## ✍️ Proposed changes

An issue came up in #leafygreen-ui about how an active SideNavItem that's wrapped by another component (such as a Tooltip or routing component) would not be detectable automatically by SideNavGroup. The addition of a `hasActiveItem` prop allows this state to be manually overwritten in cases where it may be incorrect.

🎟 _Jira ticket:_ [PD-1308](https://jira.mongodb.org/browse/PD-1308)

## 🛠 Types of changes




## ✅ Checklist




## 🧪 How to test changes



1. Run storybook locally, and navigate to the "Simple Navigation" story under SideNav.
2. Toggle the `hasActiveItem` knob. You should see the state of the second item in the collapsed nav appear active or inactive based on the knob.
